src/coinbase/webhook.ts

```diff
@@ -0,0 +1,169 @@
+import { Webhook as WebhookModel, WebhookEventType, WebhookEventFilter } from "../client/api";
+import { Coinbase } from "./coinbase";
+import { InternalError } from "./errors";
+
+/**
+ * A representation of a Webhook,
+ * which provides methods to create, list, update, and delete webhooks that are used to receive notifications of specific events.
+ */
+export class Webhook {
+  private model: WebhookModel | null;
+
+  /**
+   * Initializes a new Webhook object.
+   *
+   * @param model - The underlying Webhook object.
+   * @throws {InternalError} If the model is not provided.
+   */
+  constructor(model: WebhookModel) {
```

Comment:
should this constructor be `private`?

Reply:
yes, it could be, converted to a `private` instead